src/chrome/settings/KeyboardShortcutInput.tsx

```diff
@@ -0,0 +1,251 @@
+      altKey && ACCELERATORS.ALT,
+      ctrlKey && ACCELERATORS.CONTROL,
+      shiftKey && ACCELERATORS.SHIFT,
+      character,
+    ].filter(Boolean);
+
+    if (typeof props.onChange === 'function') {
+      props.onChange(accelerator.join('+'));
+    }
+  }, [isShortcutValid]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  let typingTimeout: NodeJS.Timeout;
+
+  const handleFocusInput = () => {
+    if (inputRef.current !== null) {
+      inputRef.current.focus();
+    }
+  };
+
+  const isCharacterSpecial = (keycode: number) =>
+    Object.values(KEYCODES).includes(keycode);
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.which === 9) {
+      return;
+    }
+
+    event.preventDefault();
+
+    const {metaKey, altKey, ctrlKey, shiftKey} = event;
+    const character = isCharacterSpecial(event.which)
+      ? ''
+      : String.fromCharCode(event.which);
+
+    setPressedKeys({
+      metaKey,
+      altKey,
+      ctrlKey,
+      shiftKey,
+      character,
+    });
+    setIsShortcutValid(undefined);
+  };
+
+  const handleKeyUp = () => {
+    const {metaKey, altKey, ctrlKey, shiftKey, character} = pressedKeys;
+
+    clearTimeout(typingTimeout);
+    typingTimeout = setTimeout(
+      () =>
+        setIsShortcutValid(
+          ([metaKey, altKey, ctrlKey, shiftKey].includes(true) &&
+            character !== '') ||
+            [metaKey, altKey, ctrlKey, shiftKey, character].every(
+              value => !value,
+            ),
+        ),
+      500,
+    );
+  };
+
+  const handleUpdatePressedKeys = (keys: PressedKeys) => {
+    setPressedKeys(keys);
+    handleKeyUp();
+    handleFocusInput();
+    setIsShortcutValid(undefined);
+  };
+
+  const renderKeys = () => {
+    const keys = [
+      pressedKeys.metaKey && '⌘',
+      pressedKeys.altKey && '⌥',
+      pressedKeys.ctrlKey && '⌃',
+      pressedKeys.shiftKey && '⇧',
+      pressedKeys.character,
+    ].filter(Boolean);
+
+    return keys.map((key, index) => (
+      <ShortcutKeyContainer key={index}>
+        <ShortcutKey>{key}</ShortcutKey>
+      </ShortcutKeyContainer>
+    ));
+  };
+
+  return (
+    <Container>
+      <Label>{props.label}</Label>
+      <ShortcutKeysContainer
+        invalid={isShortcutValid === false}
+        onClick={handleFocusInput}>
+        {renderKeys()}
+
+        <HiddenInput
+          ref={inputRef}
+          onKeyDown={handleKeyDown}
+          onKeyUp={handleKeyUp}
+        />
+      </ShortcutKeysContainer>
+
+      <FlexColumn onClick={() => handleUpdatePressedKeys(initialPressedKeys)}>
+        <CenteredGlyph name="undo" variant="outline" />
+      </FlexColumn>
+
+      <FlexColumn
+        onClick={() =>
+          handleUpdatePressedKeys({
+            metaKey: false,
+            altKey: false,
+            ctrlKey: false,
+            shiftKey: false,
+            character: '',
+          })
+        }>
+        <CenteredGlyph name="undo" variant="outline" />
```

Comment:
![Screenshot 2020-02-23 at 16 29 45](https://user-images.githubusercontent.com/9906/75115770-c6773280-5659-11ea-9a94-b363d842657d.png)

Do you think one of these could work?

Reply:
Yeah, the `cross` will mostly definitely work, thank you!

I'll change it today in the evening.